src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -101,8 +101,12 @@ private async Task ProduceDiagnosticsAsync(
                 ArrayBuilder<DiagnosticData> builder,
                 CancellationToken cancellationToken)
             {
-                var analyzersForProject = await StateManager.GetOrCreateAnalyzersAsync(project, cancellationToken).ConfigureAwait(false);
-                var hostAnalyzerInfo = await StateManager.GetOrCreateHostAnalyzerInfoAsync(project, cancellationToken).ConfigureAwait(false);
+                var solution = project.Solution;
```

Comment:
>var solution = project.Solution; [](http://example.com/codeflow?start=16&length=32)

nit: solution is only used to get SolutionState, so why not:

var solution = project.Solution.SolutionState;
var project = project.State;

Reply:
That was a dumb question, I now see after I typed it out.